native/core/src/parquet/objectstore/s3.rs

```diff
@@ -107,6 +111,37 @@ pub fn create_store(
     Ok((Box::new(object_store), path))
 }
 
+/// Get the bucket region using the [HeadBucket API]. This will fail if the bucket does not exist.
+///
+/// [HeadBucket API]: https://docs.aws.amazon.com/AmazonS3/latest/API/API_HeadBucket.html
+pub async fn resolve_bucket_region(bucket: &str) -> Result<String, Box<dyn Error>> {
+    let endpoint = format!("https://{bucket}.s3.amazonaws.com");
+    let client = reqwest::Client::new();
+
+    let response = client.head(&endpoint).send().await?;
```

Comment:
In the object store version of this code, any `reqwest` errors were consumed silently here. This version simply calls `?` and returns the underlying error.